workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.test.ts

```diff
@@ -0,0 +1,124 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "../Iterator.prototype.filter";
+import "../Iterator.prototype.map";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).toSet;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.toSet", () => {
+  it("adds primitives to a Set", () => {
+    const arr = [1, 2, 2, 3, 1, 2, 3, 3, 100, false, 100, false, "hello"];
+    const iteratorToSet = arr.values().toSet();
+
+    expect(iteratorToSet).toBeInstanceOf(Set);
+    expect(iteratorToSet).toStrictEqual(
+      new Set([1, 2, 3, 100, false, "hello"]),
+    );
+  });
+
+  it("adds objects to a Set", () => {
+    const obj1 = { a: 1 };
+    const obj2 = { b: 2 };
+    const objectArray = [obj1, obj2, obj1, obj2, obj2, obj1];
+    const iteratorToSet = objectArray.values().toSet();
```

Comment:
It's a set at this point 🙂 

```suggestion
    const set = objectArray.values().toSet();
```